src/hotspot/share/services/memReporter.cpp

```diff
@@ -161,12 +181,12 @@ void MemSummaryReporter::report() {
     MallocMemory* malloc_memory = _malloc_snapshot->by_type(flag);
     VirtualMemory* virtual_memory = _vm_snapshot->by_type(flag);
 
-    report_summary_of_type(flag, malloc_memory, virtual_memory);
+    report_summary_of_type(flag, malloc_memory, virtual_memory, read_only_bytes);
   }
 }
 
 void MemSummaryReporter::report_summary_of_type(MEMFLAGS flag,
-  MallocMemory*  malloc_memory, VirtualMemory* virtual_memory) {
+  MallocMemory*  malloc_memory, VirtualMemory* virtual_memory, size_t read_only_bytes) {
```

Comment:
You can actually move the retrieval of CDS read_only_bytes into this function now. No need to expand the interface. This function already retrieves a number of things, see e.g. thead stack printing, so there is a precedence.